opus/application/apps/cart/views.py

```diff
@@ -965,25 +1007,41 @@ def _get_download_info(product_types, session_id):
     total_download_count = 0
 
     for res in results:
-        (category, sort_order, short_name, full_name,
-         download_size, download_count, product_count) = res
+        (category, sort_order, short_name, version_name, full_name,
+         checked, download_size, download_count, product_count) = res
+        short_name_ver = short_name + '@' + version_name.lower()
         download_size = int(download_size)
         download_count = int(download_count)
         product_count = int(product_count)
-        if product_types == ['all'] or short_name in product_types:
+
+        # Check if the files info of a product type should be added up to total download
+        # info
+        is_adding_up_to_total = False
+        for p in product_types:
+            if '@' in p:
+                prod_type, _, p_version = p.partition('@')
+                if p_version.lower() == 'current':
+                    p_version = 'current'
+                if short_name == prod_type and version_name.lower() == p_version:
+                    is_adding_up_to_total = True
+                    break
+            elif short_name == p:
+                is_adding_up_to_total = True
+                break
+        if (product_types == ['all'] and checked) or is_adding_up_to_total:
```

Comment:
This doesn't seem right to me. _get_download_info should not be checking whether an item is checked by default. "all" should really mean all. Perhaps we need a function flag that says something like "count_default_checked_only"? Or is there a way to pass in the list of default-checked product types?

Reply:
I thought about this some more. Perhaps this is OK after all, since it's only for internal use. But there needs to be a comment added to the function header saying that "all" doesn't really mean ALL.

Reply:
I think what you want to say is:
The resulting totals are limited to the given product_types. ['all'] means include all product types that are "checked by default" in the database.